### PR TITLE
Single vmlinux.h File for examples

### PR DESCRIPTION
**Pull request by mimullin-bbry**
Create a new vmlinux.h file from Kernel 5.14 called
    examples/vmlinux_514.h.  Have all examples use examples/vmlinux_514.h
    as their vmlinux.h, reducing the amount of vmlinux's carried by the
    project.

    Fix an issue in runqslower due to task_struct::state changing to
    task_struct::__state.

    Signed-off-by: Michael Mullin <mimullin@blackberry.com>

### CODE REVIEW

File: libbpf-rs/src/program.rs [hunk 1]
```diff
@@ -73,6 +73,8 @@ pub enum ProgramType {
     StructOps,
     Ext,
     Lsm,
+    SkLookup,
```
Comment (mimullin-bbry) [inline libbpf-rs/src/program.rs:76]: I do not know why this code was pulled into my PR.  I did not edit anything in program.rs.

Will this be a problem?

Comment (anakryiko) [reply libbpf-rs/src/program.rs:76]: yeah, you have 3 extra commits. You need to pull latest master, rebase your commit on top of it and force-push your branch.

**Comment by insearchoflosttime** [comment]
Thanks for working on this. Would you mind splitting your commit into 2 (one for the vmlinux.h change and one for the runqslower change)?

File: examples/runqslower/src/bpf/runqslower.bpf.c [hunk 2]
```diff
@@ -1,9 +1,15 @@
 // SPDX-License-Identifier: GPL-2.0
 // Copyright (c) 2019 Facebook
+
 #include "vmlinux.h"
 #include <bpf/bpf_helpers.h>
+#include <bpf/bpf_core_read.h>
 #include "runqslower.h"
 
+struct task_struct_pre_5_14 {
```
Comment (anakryiko) [inline examples/runqslower/src/bpf/runqslower.bpf.c:9]: this won't work, you need to use three underscored after `task_struct` to make libbpf recognize it as just another flavor of task_struct

File: examples/runqslower/src/bpf/runqslower.bpf.c [hunk 3]
```diff
@@ -62,6 +68,15 @@ int handle__sched_wakeup_new(u64 *ctx)
 	return trace_enqueue(p->tgid, p->pid);
 }
 
+static inline
+long get_task_state(struct task_struct *t)
```
Comment (anakryiko) [inline examples/runqslower/src/bpf/runqslower.bpf.c:72]: keep on single line?

File: examples/runqslower/src/bpf/runqslower.bpf.c [hunk 4]
```diff
@@ -1,9 +1,15 @@
 // SPDX-License-Identifier: GPL-2.0
 // Copyright (c) 2019 Facebook
+
 #include "vmlinux.h"
 #include <bpf/bpf_helpers.h>
+#include <bpf/bpf_core_read.h>
 #include "runqslower.h"
 
+struct task_struct_pre_5_14 {
+	volatile long int state;
```
Comment (anakryiko) [inline examples/runqslower/src/bpf/runqslower.bpf.c:10]: nit: volatile not really necessary, you are going to be bpf_probe_read()'ing it either way

**Review by anakryiko — COMMENTED** [review]
Please make sure that your commit subject fits single line and is shorter than 80 characters. After that should be empty line and then commit description.

**Pull request merged**